### ISSUE TITLE
[BUG] App crash on StateContainer state change

### ISSUE THREAD

**Issue by thombrink** [CommunityToolkit/Maui#825]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When changing the state of a StateContainer to quickly, the app crashes with the following exception (on iOS):

`[0:] An error occurred: 'A task was canceled.'. Callstack: '   at CommunityToolkit.Maui.Layouts.StateContainerController.FadeLayoutChildren(Layout layout, Boolean shouldAnimate, Boolean isHidden, CancellationToken token) in /_/src/CommunityToolkit.Maui/Layouts/StateContainer/StateContainerController.shared.cs:line 146
   at CommunityToolkit.Maui.Layouts.StateContainerController.SwitchToState(String state, Boolean shouldAnimate, Nullable`1 cancellationToken) in /_/src/CommunityToolkit.Maui/Layouts/StateContainer/StateContainerController.shared.cs:line 124
   at CommunityToolkit.Maui.Layouts.StateContainer.OnCurrentStateChanged(BindableObject bindable, Object oldValue, Object newValue) in /_/src/CommunityToolkit.Maui/Layouts/StateContainer/StateContainer.shared.cs:line 84
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at Foundation.NSAsyncSynchronizationContextDispatcher.Apply()
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName)
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass)
   at Eqontrol.Maui.Program.Main(String[] args) in C:\Users\Thom\source\repos\Eqontrol\Backend\Eqontrol.Maui\Platforms\iOS\Program.cs:line 13
   at System.Reflection.MethodInvoker.InterpretedInvoke(Object obj, Span`1 args, BindingFlags invokeAttr)'
The app has been terminated.`

### Expected Behavior

Should transition to the new state.

### Steps To Reproduce

1. Create a StateContainer
2. Add a gesture recognizer to it with a command that changes the state binding value
3. Tab on the StateContainer and tab again before the new state is fully visible

### Link to public reproduction project repository

http://no-time-to-reproduce-right.now

### Environment

```markdown
- .NET MAUI CommunityToolkit: 3.0.0 (previous versions also had the same behaviour)
- OS: Windows 11
- .NET MAUI: iOS
```

**Comment by thombrink** [comment]
Possibly related to #806

**Comment by jfversluis** [comment]
Hey @thombrink!

I noticed that you have a very creative link for the reproduction, unfortunately it doesn't seem to exist. Could you please still add it? Because like you don't have time to create a reproduction, that basically makes your problem our problem by potentially spending a lot more time trying to read your mind and guess what your code might look like.

I think it will both save us a lot of time and frustration if you could still get us that reproduction so we know we're looking at the right thing and we can diagnose and potentially fix as soon as possible. Or, of course, feel free to open a PR if you still need a project over the holidays.

Thanks!

**Comment by thombrink** [comment]
Sure! I could have been more descriptive, but I meant I wanted to report it and create a reproduction when I'm in a position to/ have some free time.

Sorry for the inconvenience!

P.S. I'm sure Edgar is oke with it, as I'm no longer working for him ;) I also hope to deliver a maui hybrid js example app soon, so stay tuned!

**Comment by jfversluis** [comment]
No inconvenience my friend! Sorry if I came across a bit strong, there unfortunately are a lot people less helpful. That wasn't my intention, so all good!

Haha alright! Just noticed 4DotNet on your GitHub profile which I found funny, blast from the past 😉 Looking forward to your app, feel free to reach out and share. Always cool to see something!

**Comment by thombrink** [comment]
Haha never mind! You just #triggerd me a bit (positively), so have a look at [the fix branch over here](https://github.com/CommunityToolkit/Maui/compare/main...thombrink:CommunityToolkitMaui:patch/fade-animation-fix) note **UNTESTED CODE** hope to test it somewere this week and create a PR afterwards.

Yeah I know, we just missed each other over there. Be prepared for some awesome code haha (disclaimer: it's just a simple navigation wrapper/service to navigate to and interact with js pages inside a webview over a local websocket :).

**Comment by brminnick** [comment]
Fixed in #811
